packages/core/src/Stores/client-store.js

```diff
@@ -229,21 +229,38 @@ export default class ClientStore extends BaseStore {
 
     @computed
     get legal_allowed_currencies() {
-        if (!this.landing_companies || !this.root_store.ui) return [];
-        if (!this.root_store.ui.real_account_signup_target) {
+        const getDefaultAllowedCurrencies = () => {
             if (this.landing_companies.gaming_company) {
                 return this.landing_companies.gaming_company.legal_allowed_currencies;
             }
             if (this.landing_companies.financial_company) {
                 return this.landing_companies.financial_company.legal_allowed_currencies;
             }
             return [];
+        };
+
+        if (!this.landing_companies || !this.root_store.ui) return [];
+        if (!this.root_store.ui.real_account_signup_target) {
+            getDefaultAllowedCurrencies();
         }
         if (['set_currency', 'manage'].includes(this.root_store.ui.real_account_signup_target)) {
             return this.current_landing_company.legal_allowed_currencies;
         }
         const target = this.root_store.ui.real_account_signup_target === 'maltainvest' ? 'financial' : 'gaming';
-        return this.landing_companies[`${target}_company`].legal_allowed_currencies;
+
+        if (this.landing_companies[`${target}_company`]) {
+            return this.landing_companies[`${target}_company`].legal_allowed_currencies;
+        }
+
+        if (this.landing_companies.gaming_company) {
+            return this.landing_companies.gaming_company.legal_allowed_currencies;
+        }
+
+        if (this.landing_companies.financial_company) {
+            return this.landing_companies.financial_company.legal_allowed_currencies;
+        }
```

Comment:
we can remove this now, as this will be handled in `getDefaultAllowedCurrencies`